packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/workflow/InputPort.java

```diff
@@ -0,0 +1,23 @@
+package software.uncharted.terarium.hmiserver.models.dataservice.workflow;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.io.Serializable;
+import java.util.List;
+import java.util.UUID;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Accessors;
+
+@Data
+@Accessors(chain = true)
```

Comment:
Not urgent at all of course can be done in later PRs as this is big enough as is
If we add TSModel to these we can remove some of the typing done in `src/types/workflow.ts`

Reply:
I am purposely NOT generating the workflow types because they are still not syntactically compatible between Java and Typescript.